services/tasks/src/test/java/tech/pegasys/pantheon/services/tasks/CachingTaskCollectionTest.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.services.tasks;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import tech.pegasys.pantheon.metrics.noop.NoOpMetricsSystem;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class CachingTaskCollectionTest {
+  @ClassRule public static final TemporaryFolder folder = new TemporaryFolder();
+
+  private final TaskCollection<BytesValue> wrappedTaskCollection;
+
+  public CachingTaskCollectionTest(
+      final String testName, final TaskCollectionSupplier wrappedTaskCollection) throws Exception {
+    this.wrappedTaskCollection = wrappedTaskCollection.get();
+  }
+
+  private CachingTaskCollection<BytesValue> createCachingCollection(final int cacheSize) {
+    return new CachingTaskCollection<>(wrappedTaskCollection, cacheSize);
+  }
+
+  @Parameters(name = "Wrap {0}")
+  public static Collection<Object[]> getTestParametersForConfig() throws IOException {
```

Comment:
Why do we need to test this wrapping two different implementations? If we have tests to ensure `InMemoryTaskQueue` and `RocksDbTaskQueue` individually implement `TaskCollection` correctly this we should be able to just test the caching later with either one of them (probably `InMemoryTaskQueue` for simplicity and faster tests).

Reply:
The thought was it would give extra coverage and potentially reveal inconsistencies, but you're right - it slows down the tests and probably isn't worth it.  Updated.